LargeNetworkAnalysisTools.pyt

```diff
@@ -0,0 +1,377 @@
+        )
+
+        param_out_origins = arcpy.Parameter(
+            displayName="Output Updated Origins",
+            name="Output_Updated_Origins",
+            datatype="DEFeatureClass",
+            parameterType="Required",
+            direction="Output"
+        )
+
+        param_out_destinations = arcpy.Parameter(
+            displayName="Output Updated Destinations",
+            name="Output_Updated_Destinations",
+            datatype="DEFeatureClass",
+            parameterType="Required",
+            direction="Output"
+        )
+
+        param_network = arcpy.Parameter(
+            displayName="Network Data Source",
+            name="Network_Data_Source",
+            datatype="GPNetworkDataSource",
+            parameterType="Required",
+            direction="Input"
+        )
+
+        param_travel_mode = arcpy.Parameter(
+            displayName="Travel Mode",
+            name="Travel_Mode",
+            datatype="NetworkTravelMode",
+            parameterType="Required",
+            direction="Input"
+        )
+        param_travel_mode.parameterDependencies = [param_network.name]
+
+        param_time_units = arcpy.Parameter(
+            displayName="Time Units",
+            name="Time_Units",
+            datatype="GPString",
+            parameterType="Required",
+            direction="Input"
+        )
+        param_time_units.filter.list = odcm.TIME_UNITS
+        param_time_units.value = "Minutes"
+
+        param_distance_units = arcpy.Parameter(
+            displayName="Distance Units",
+            name="Distance_Units",
+            datatype="GPString",
+            parameterType="Required",
+            direction="Input"
+        )
+        param_distance_units.filter.list = odcm.DISTANCE_UNITS
+        param_distance_units.value = "Miles"
+
+        param_chunk_size = arcpy.Parameter(
+            displayName="Maximum Origins and Destinations per Chunk",
+            name="Max_Inputs_Per_Chunk",
+            datatype="GPLong",
+            parameterType="Required",
+            direction="Input"
+        )
+        param_chunk_size.value = 1000
+
+        param_max_processes = arcpy.Parameter(
+            displayName="Maximum Number of Parallel Processes",
+            name="Max_Processes",
+            datatype="GPLong",
+            parameterType="Required",
+            direction="Input"
+        )
+        param_max_processes.value = 4
+
+        param_cutoff = arcpy.Parameter(
+            displayName="Cutoff",
+            name="Cutoff",
+            datatype="GPDouble",
+            parameterType="Optional",
+            direction="Input"
+        )
+
+        param_num_dests = arcpy.Parameter(
+            displayName="Number of Destinations to Find for Each Origin",
+            name="Num_Destinations",
+            datatype="GPLong",
+            parameterType="Optional",
+            direction="Input"
+        )
+
+        param_barriers = arcpy.Parameter(

```

Comment:
@mmorang 
I had to look through the code to figure out that barriers parameter is supporting all three barrier types. Can we somehow make this more obvious? An easy way would have been if we have side panel help for this parameter.

Reply:
I added side panel help.